src/GHCMod/Options/Commands.hs

```diff
@@ -0,0 +1,230 @@
+module GHCMod.Options.Commands where
+
+import Options.Applicative
+import Options.Applicative.Types
+import Language.Haskell.GhcMod.Types
+import GHCMod.Options.DocUtils
+import Language.Haskell.GhcMod.Read
+
+type Symbol = String
+type Expr = String
+type Module = String
+type Line = Int
+type Col = Int
+type Point = (Line, Col)
+
+data GhcModCommands =
```

Comment:
As @alanz says, encoding commands as datatype actually makes some sense in the context of making interactive commands more type-safe and, generally speaking, more reliable.

For example, I could make a new interactive mode work just by `read`ing `GhcModCommands` from stdin. From my point of view, this is a reason enough to fiddle with additional datatype.